excercsises/1_introduction/minimal.spec.js

```diff
@@ -43,15 +58,33 @@ describe('Protractor workshop app', function() {
 	 */
 
 	it('should have Feature A, Feature B, Feature C sections ...', function(){
-		var features = element.all(by.xpath('//h2'));
+		var features = element.all(by.xpath('//h2')).map(function(elm, index) {
+			return {
+				index: index,
+				text: elm.getText(),
+			}
+		});
+		var expectedAtr = [
+			{index: 0, text: 'Feature A'},
+			{index: 1, text: 'Feature B'},
+			{index: 2, text: 'Feature C'}
+		]
+		expect(features).toEqual(expectedAtr);
 	});
 
 	/**
 	 * Check http://angular.github.io/protractor/#/api?view=ElementArrayFinder
 	 * to see how fileter function can be used select elements based on condition
 	 */	
 	it('should route to "Blog" pages after selecting link',function(){
-		var menuItems = element.all(by.css('ul.nav > li > a'));
+		var menuItems = element.all(by.css('ul.nav > li > a')).filter(function(elem, index) {
+			return elem.getText().then(function(text) {
+				return text === 'Blog';
+			});
+		}).first().click();
+		var newPageHeader = element(by.className('page-header')).element(by.tagName('h1'));
```

Comment:
not that it is wrong but you can also do it like this to e.g. check the current URL.
```
let pageTitle = browser.driver.getTitle();
expect(pageTitle).toBe('Protractor workshop | Blog');
```